content/fast/adaptive-serving-based-on-network-quality/index.md

```diff
@@ -0,0 +1,103 @@
+
+One way to deal with this is by adapting which assets you're serving to users based on the quality of their connection. This is now [possible](https://caniuse.com/#feat=netinfo) with the [Network Information API](https://developer.mozilla.org/en-US/docs/Web/API/Network_Information_API) which enables web applications to access information about the user's network.
+
+## Usage
+
+There are many ways you can use this network information to improve the user experience:
+
+* Switch between serving high-definition and low-definition content based on the user's network. 
+* Decide whether to preload resources. 
+* Defer uploads and downloads when users are on a slow connection. 
+* Enable offline mode if the network quality is not good enough to load the app and use the features.
+* Warn users that doing something (e.g. watching video) over cellular could cost them money.
+* Use it in your analytics to gather data on your users’ network quality.
+
+Many applications are already doing something similar. For example, YouTube, Netflix and most other video (or video calling) services automatically adjust the resolution during streaming. When Gmail is loading, it provides users with a link to "load basic HTML (for slow connections)" - a simplified (yet functional) version of Gmail. 
+
+<img class="screenshot" src="./loading_gmail_slow_connection.png" alt="A link to load basic HTML version of Gmail when users are on slow connections">
+
+With the Network Information API you can now implement these functionalities in your own applications without much effort.
+
+## How it works
+
+The **navigator.connection** object contains information about a client’s connection. Its properties are:
+
+1. **downlink**: the bandwidth estimate in megabits per second
+2. **effectiveType**: the effective type of the connection, with possible values **‘slow-2g’**,**‘2g’**, **‘3g’**, or **‘4g’**
+3. **onchange:** an event handler that fires when connection information changes
+4. **rtt**: the estimated round-trip latency of the connection in milliseconds
+5. **saveData:** boolean that defines whether the user has requested a reduced data usage mode 
+
+Here’s what this looks like when you run it in the browser’s console:
+
+<img class="screenshot" src="./network_information_in_console.jpg" alt="Chrome DevTools console displaying the values of navigator.connection object's properties">
+
+It might seem odd that the **effectiveType **only goes up to ‘**4g’**, but that segment actually includes all connections that are **‘4g’ and higher.** The API determines the effective connection type based on the combination of **round-trip time (RTT) and downlink speed**. For example, if you have a 10 Mbps downlink combined with 1000 ms latency, the **effectiveType **will be ‘3g’ due to the latency. 
```

Comment:
As mentioned above, I would suggest moving the description of how effectiveType is determined up to it's definition. I'd also suggest making the language as simple as possible, since it is in a table. Something like: Based on combination of round-trip time and downlink speed. For example, a 10 Mbps downlink combined with 1000ms latency is '3g' due to latency.

Reply:
I'm not entirely sure I understand how effectiveType is calculated, in terms of combinations. Can you have a low RTT and low download link, and if so, how does that combination get calculated? Still isn't quite clear to me-- the calculation aspect of the API.

Reply:
Rewrote this to "For example, fast downlink combined with high latency will have lower effectiveType due to latency.". Didn't mention any numbers since I've removed the table that references the exact thresholds for calculation. That calculation aspect tripped me up in the beginning too. I was very confused when I would get 3g effectiveType on my laptop at home, but I figured out from the table that it was happening because of the high latency. (That's why I included the table in the first place.)

I think RTT && downlink relationship can be summed up with 'A chain is only as strong as its weakest link'. The worse value of the two will determine the effectiveType.  Low RTT (good) && low downlink (bad) in combination will have the effectiveType that corresponds to the poor downlink value. You'll get '4g' only if both of the values are in the '4g' range from the table.

I hope this helps and please let me know if you think this can be phrased better in the doc :)

Reply:
I think the rewrite is a lot more clear personally - interested to hear what @Meggin thinks :)